### PR TITLE
adding babel plugin for `IS_DEV` and `IS_MINIFIED` transformation

### LINKED ISSUE #19618: Single Pass: Add Constants transforms
To fully get rid of our custom runner for single pass we need to move any transformation it does to a babel transform.

IS_DEV variable -> always `false` if `--fortesting` flag is not used with `gulp dist` command
IS_MINIFIED variable -> always `true`. (this is because if gulp dist is used, the js we output is assumed as minified)

these variables can be found in `src/mode.js`. We also enforce in a presubmit rule that `IS_DEV` is only used in this file.

### PR DESCRIPTION
**Pull request by prateekbh**
Fixes #19618
- Adds a babel transform which sets `IS_DEV` to false and `IS_MINIFIED` to true if `--isfortesting` is not present

### CODE REVIEW

File: build-system/babel-plugins/babel-plugin-is_minified-constant-transformer/index.js [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+ * limitations under the License.
+ */
+
+/**
+  * Changes the values of IS_DEV to false and IS_MINIFIED to true.
+  * The above said variables are in src/mode.js file.
+  * @param {Object} babelTypes
+  */
+module.exports = function(babelTypes) {
+  const {types: t} = babelTypes;
```
Comment (jridgewell) [inline build-system/babel-plugins/babel-plugin-is_minified-constant-transformer/index.js]: Nit: inline this into the parameter

File: build-system/babel-plugins/babel-plugin-is_dev-constant-transformer/index.js [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+  */
+module.exports = function(babelTypes) {
+  const {types: t} = babelTypes;
+  return {
+    visitor: {
+      VariableDeclarator(path) {
+        const {id, init} = path.node;
+        if (t.isIdentifier(id, {name: 'IS_DEV'})
+            && t.isBooleanLiteral(init, {value: true})) {
+          path.replaceWith(
```
Comment (jridgewell) [inline build-system/babel-plugins/babel-plugin-is_dev-constant-transformer/index.js]: Just replace the init.

Comment (prateekbh) [reply build-system/babel-plugins/babel-plugin-is_dev-constant-transformer/index.js]: I actually tried doing that...
can you please explain why line 73 is still true after this transform 

https://astexplorer.net/#/gist/e39e874a4c808c16842a259c850df995/87476cd3629dc264148bd0b40a2daa95b83bce3d

Comment (jridgewell) [reply build-system/babel-plugins/babel-plugin-is_dev-constant-transformer/index.js]: https://astexplorer.net/#/gist/e39e874a4c808c16842a259c850df995/b890b7f1668f81110bd8e64b4a4e053f60f67368

Comment (kristoferbaxter) [reply build-system/babel-plugins/babel-plugin-is_dev-constant-transformer/index.js]: Replacing the path isn't a bad idea, but if you want to get around the issue, change the following in your snippet.

`const {id, init} = node;` => `let {id, init} = node;`

**Pull request merged**